src/core_plugins/metrics/public/lib/validate_interval.js

```diff
@@ -0,0 +1,16 @@
+import parseInterval from 'ui/utils/parse_interval';
+export function validateInterval(timefilter, panel, maxBuckets) {
+  const { interval } = panel;
+  const { min, max } = timefilter.getBounds();
+  // No need to check auto it will return around 100
+  if (interval === 'auto') return;
+  const duration = parseInterval(interval);
+  if (!duration) {
+    throw new Error(`Invalid interval: ${interval} is not a valid interval`);
+  }
+  const span = max.valueOf() - min.valueOf();
+  const buckets = span / duration.asMilliseconds();
```

Comment:
Math.floor the buckets. Otherwise we get some googly error messages :)

![image](https://cloud.githubusercontent.com/assets/1833023/25114371/4338aa42-23cc-11e7-9b74-14aede24181a.png)